woocommerce/woocommerce
# Analytics API: Search for customers using all available search fields instead of having to choose one

### Changes proposed in this Pull Request:

One of the frustrations with customer search in the API has been that you have to specify which field you're searching in (i.e. name, username, or email). Most times if you're providing a search string, you'd want to be able to search in all of those fields at once. With this, now you can.

Refs #33383

### How to test the changes in this Pull Request:

1. Ensure that the Analytics feature is enabled (WooCommerce > Settings > Advanced > Features).
1. Create two new orders with different customer data. Make sure the name of one customer has a substring in common with the other customer's email address.
2. Manually create a customer user account (Users > Add new) and make sure the user name includes the same substring that the other two customers have. Then create a new order and assign it to the new customer user.
3. Use an API client like Postman or Insomnia, or use something like cURL to make requests to the `wc-analytics/reports/customers` endpoint:
    * Include the `search` parameter with your common substring, and include the `searchby` parameter with the value `all`.
    * Example request:
      ```
      curl --request GET \
      --url 'http://localhost:8888/wp-json/wc-analytics/reports/customers?search=herb&searchby=all' \
      --header 'Authorization: Basic {your app password}'
      ```
4. Ensure that the response has the expected customer records in it, but not any others. Try using other `search` strings, and other `searchby` values (`name`, `username`, `email`)
